open-api/lang-tools/src/main/java/io/deephaven/lang/completion/ChunkerCompleter.java

```diff
@@ -1018,9 +1018,12 @@ private Optional<Class<?>> resolveScopeType(
 
         IsScope o = scope.get(0);
 
-        final Class<?> type = variables.getVariableType(o.getName());
-        if (type != null) {
-            return Optional.of(type);
+        final Object instance = variables.readParamValue(o.getName(), null);
+        if (instance != null) {
+            if (instance instanceof Table) {
+                return Optional.of(Table.class);
```

Comment:
I don't get this. Why are we restricting sub-classes of Table to the top-level interface? Does it solve a real problem?

Reply:
This is inlining the old implementation that VariableProvider did, replicated without additional thought.

Reply:
I would want to think long and hard before pushing `BaseTable` or `QueryTable` methods into auto-complete, to be honest. Let's not change the status quo in this PR.